mutation_partition.hh

```diff
@@ -826,6 +826,7 @@ public:
 
     void apply(tombstone deleted_at) {
         _deleted_at.apply(deleted_at);
+        maybe_shadow();
```

Comment:
I didn't understand how maybe_shadow() checks if the shadowable tombstone is shadowed by the regular one. Doesn't it check if the shadowable tombstone is shadowed by the marker?